File: llm_generated_code/Gemini/CPP/Level1/Q28.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string word;
    cin >> word;

    int upperCount = 0, lowerCount = 0;
    for (char c : word) {
        if (isupper(c)) {
            upperCount++;
        } else {
            lowerCount++;
        }
    }

    if (lowerCount >= upperCount) {
        transform(word.begin(), word.end(), word.begin(), ::tolower);
    } else {
        transform(word.begin(), word.end(), word.begin(), ::toupper);
    }

    cout << word << endl;

    return 0;
}

File: llm_generated_code/Gemini/Python/Level2/Q28.py
def normalize_word(word):
    upper_count = sum(1 for c in word if c.isupper())
    lower_count = len(word) - upper_count
    if upper_count > lower_count:
        return word.upper()
    else:
        return word.lower()

def main():
    word = input()
    print(normalize_word(word))

if __name__ == "__main__":
    main()
